### PR TITLE
ISPN-6725 Enhance ContinuousQueryListener to receive events for updat…

### PR DESCRIPTION
**Pull request by anistor**
…ed results

https://issues.jboss.org/browse/ISPN-6725

### CODE REVIEW

**Comment by anistor** [comment]
Updated.

**Comment by slaskawi** [comment]
Could you please rebase it? After integrating https://github.com/infinispan/infinispan/pull/4387 there should be only one commit here (if I'm not mistaken).

File: remote-query/remote-query-client/src/main/java/org/infinispan/query/remote/client/MarshallerRegistration.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ private MarshallerRegistration() {
     *
     * @param ctx the serialization context
     * @throws org.infinispan.protostream.DescriptorParserException if a proto definition file fails to parse correctly
-    * @throws IOException if proto file registration fails
+    * @throws IOException                                          if proto file registration fails
```
Comment (slaskawi) [inline remote-query/remote-query-client/src/main/java/org/infinispan/query/remote/client/MarshallerRegistration.java]: Probably some formatting problem

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/event/ContinuousQueryImpl.java [hunk 2]
```diff
@@ -10,9 +10,9 @@
 import org.infinispan.client.hotrod.marshall.ProtoStreamMarshaller;
 import org.infinispan.protostream.ProtobufUtil;
 import org.infinispan.protostream.SerializationContext;
-import org.infinispan.query.dsl.Query;
```
Comment (slaskawi) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/event/ContinuousQueryImpl.java]: Probably unnecessary import reordering...

File: query-dsl/src/main/java/org/infinispan/query/api/continuous/ContinuousQueryListener.java [hunk 3]
```diff
@@ -14,12 +14,23 @@
     * @param key   the key of the joining entry
     * @param value the joining entry or the Object[] projection if specified
     */
-   void resultJoining(K key, V value);
+   default void resultJoining(K key, V value) {
```
Comment (slaskawi) [inline query-dsl/src/main/java/org/infinispan/query/api/continuous/ContinuousQueryListener.java]: Maybe we should deprecate them?

File: query/src/test/java/org/infinispan/query/continuous/ContinuousQueryProfilingTest.java [hunk 4]
```diff
@@ -4,10 +4,10 @@
 import org.infinispan.configuration.cache.CacheMode;
 import org.infinispan.configuration.cache.ConfigurationBuilder;
 import org.infinispan.query.Search;
-import org.infinispan.query.dsl.Query;
-import org.infinispan.query.dsl.QueryFactory;
 import org.infinispan.query.api.continuous.ContinuousQuery;
```
Comment (slaskawi) [inline query/src/test/java/org/infinispan/query/continuous/ContinuousQueryProfilingTest.java]: import reordering

**Comment by slaskawi** [comment]
Changes look good to me. There are some imports reordering which should be fixed once we merge https://github.com/infinispan/infinispan/pull/4439. 

The only thing that I'm wondering is whether we should (or should not) deprecate `resultJoining` and `resultLeaving`?

**Comment by anistor** [comment]
resultJoining / resultLeaving will not be deprecated. The empty default implementation is just meant to save a few keystrokes if you are not interested in processing those events.

File: client/hotrod-client/src/main/java/org/infinispan/client/hotrod/event/ContinuousQueryImpl.java (hunk 2 above)
Comment (anistor) [inline client/hotrod-client/src/main/java/org/infinispan/client/hotrod/event/ContinuousQueryImpl.java]: The imports are supposed to be ordered alphabetically.

File: remote-query/remote-query-client/src/main/java/org/infinispan/query/remote/client/MarshallerRegistration.java (hunk 1 above)
Comment (anistor) [inline remote-query/remote-query-client/src/main/java/org/infinispan/query/remote/client/MarshallerRegistration.java]: Fixed.

**Comment by anistor** [comment]
Rebased.

File: query/src/test/java/org/infinispan/query/continuous/ContinuousQueryTest.java [hunk 5]
```diff
@@ -21,6 +15,13 @@
 import org.infinispan.util.TimeService;
 import org.testng.annotations.Test;
 
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static org.infinispan.query.dsl.Expression.max;
+import static org.infinispan.query.dsl.Expression.param;
+import static org.junit.Assert.assertEquals;
```
Comment (danberindei) [inline query/src/test/java/org/infinispan/query/continuous/ContinuousQueryTest.java]: Now here's an unnecessary import reordering -- the code style from https://github.com/infinispan/infinispan/pull/4439 says static imports should be on top.

**Comment by anistor** [comment]
I've reordered the imports in all affected files according to the code style.

**Comment by anistor** [comment]
And rebased.

**Comment by danberindei** [comment]
Integrated, thanks Adrian!

**Comment by anistor** [comment]
Thanks Dan. The jira is still open. I'll close it myself.

**Pull request closed without merging**